tests/integration/lifecycle/test_craftctl.py

```diff
@@ -0,0 +1,428 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2021 Canonical Ltd.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License version 3 as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import os
+import sys
+import textwrap
+from pathlib import Path
+
+import pytest
+import yaml
+
+import craft_parts
+from craft_parts import Action, Step, errors
+from tests import TESTS_DIR
+
+
+@pytest.fixture(autouse=True)
+def setup_fixture(new_dir, mocker):
+    craftctl = Path("craftctl")
+    craftctl.write_text(f"#!{sys.executable}\nfrom craft_parts import ctl\nctl.main()")
+    craftctl.chmod(0o755)
+
+    mocker.patch.dict(
+        os.environ,
+        {
+            "PATH": "/bin" + os.pathsep + str(new_dir),
+            "PYTHONPATH": str(TESTS_DIR.parent),
+        },
+    )
+
+    mocker.patch("craft_parts.utils.os_utils.mount")
+    mocker.patch("craft_parts.utils.os_utils.mount_overlayfs")
+    mocker.patch("craft_parts.utils.os_utils.umount")
+
+
+def test_craftctl_default(new_dir, capfd, mocker):
```

Comment:
this is a large test, a code comment in each block describing the intention of the assertions would be nice

Reply:
Added comments to blocks.